src/approaches/nsrt_rl_approach.py

```diff
@@ -49,33 +65,215 @@ def get_interaction_requests(self) -> List[InteractionRequest]:
+                option_objects_unchanged = np.allclose(
+                    prev_state_unchanging_feat - curr_state_unchanging_feat,
+                    0,
+                    atol=1e-5)
+                other_objects_unchanged = np.allclose(
+                    curr_states[-2].vec(other_objects),
+                    curr_states[-1].vec(other_objects),
+                    atol=1e-5)
+                if not option_objects_unchanged or not other_objects_unchanged:
+                    reward += self._neg_reward * 10
+
+                # Add pos_reward if we got within epsilon of the option's
+                # subgoal, otherwise we add neg_reward.
+                # Ignore the optimization that checks for repeated states, which
+                # messes with the terminal state check.
+                _ = curr_option.memory.pop("last_state", None)
+                if curr_option.terminal(s):
+                    # Check if we reached our subgoal within a tolerance.
+                    if np.allclose(relative_param,
+                                   0,
+                                   atol=self._reward_epsilon):
+                        curr_rewards.append(reward + self._pos_reward)
+                    else:
+                        curr_rewards.append(reward + self._neg_reward)
+
+                    # Store transition data.
+                    option_to_data[curr_option.name].append(
+                        (curr_states, curr_actions, curr_rewards,
+                         curr_relative_params))
+
+                    # Advance to next option.
+                    curr_option_idx += 1
+                    if curr_option_idx < len(plan):
+                        curr_option = plan[curr_option_idx]
+                        parent = cast(_LearnedNeuralParameterizedOption,
+                                      curr_option.parent)
+                        var_to_obj = dict(
+                            zip(
+                                parent._operator.parameters,  # pylint: disable=protected-access
+                                curr_option.objects))
+                        var_to_unchanging_feat_ind = \
+                            NSRTReinforcementLearningApproach._get_unchanging_features(  # pylint: disable=line-too-long
+                            parent._changing_var_to_feat, var_to_obj)  # pylint: disable=protected-access
+                        if curr_option.name not in option_to_data:
+                            option_to_data[curr_option.name] = []
+                    else:
+                        # If we run out of options in the plan, there should be
+                        # an _OptionPlanExhausted exception, and so there is
+                        # nothing more in the trajectory that we have not yet
+                        # assigned to an _Option already. That is, the current
+                        # state in this loop is the last state.
+                        assert s == traj.states[-1]
```

Comment:
state equality is not defined. if you want to assert that this is the same python object, use `is` instead of `==`. if you just want to check that the states are semantically equivalent, use `state.allclose(other)`